### ISSUE TITLE
amplify_output.json not updated (updated async after a day) after each deployment

### ISSUE THREAD

**Issue by scorobogaci** [aws-amplify/amplify-hosting#4039]
### Environment information

```plain text
System:
  OS: macOS 14.5
  CPU: (8) arm64 Apple M1
  Memory: 62.39 MB / 16.00 GB
  Shell: /bin/zsh
Binaries:
  Node: 20.15.0 - ~/.nvm/versions/node/v20.15.0/bin/node
  Yarn: 1.22.19 - /opt/homebrew/bin/yarn
  npm: 10.7.0 - ~/.nvm/versions/node/v20.15.0/bin/npm
  pnpm: undefined - undefined
NPM Packages:
  @aws-amplify/auth-construct: 1.3.1
  @aws-amplify/backend: 1.3.0
  @aws-amplify/backend-auth: 1.2.0
  @aws-amplify/backend-cli: 1.2.8
  @aws-amplify/backend-data: 1.1.4
  @aws-amplify/backend-deployer: 1.1.4
  @aws-amplify/backend-function: 1.5.0
  @aws-amplify/backend-output-schemas: 1.2.0
  @aws-amplify/backend-output-storage: 1.1.2
  @aws-amplify/backend-secret: 1.1.3
  @aws-amplify/backend-storage: 1.2.0
  @aws-amplify/cli-core: 1.1.3
  @aws-amplify/client-config: 1.3.2
  @aws-amplify/deployed-backend-client: 1.4.1
  @aws-amplify/form-generator: 1.0.3
  @aws-amplify/model-generator: 1.0.8
  @aws-amplify/platform-core: 1.1.0
  @aws-amplify/plugin-types: 1.3.0
  @aws-amplify/sandbox: 1.2.2
  @aws-amplify/schema-generator: 1.2.4
  aws-amplify: 6.6.2
  aws-cdk: 2.159.1
  aws-cdk-lib: 2.159.1
  typescript: 5.6.2
AWS environment variables:
  AWS_STS_REGIONAL_ENDPOINTS = regional
  AWS_NODEJS_CONNECTION_REUSE_ENABLED = 1
  AWS_SDK_LOAD_CONFIG = 1
No CDK environment variables
```

### Describe the bug

When using Amplify Gen 2 , and making some updates in `data/resource.ts` (AppSync schema), the updates on models, configurations are not reflected in amplify_output.json after the deployment succeeded.
Even though the logs (deployment logs) shows
`[INFO]: File written: amplify_outputs.json`
in fact latest changes are not written there.

Funny fact, if you download the `amplify_output.json` the next day, changes are reflected ))) The most hilarious bug ever seen I guess (some async things going on)

Here is the backend build in `amplify.yml`
```
backend:
  phases:
    build:
      commands:
        - npm ci --cache .npm --prefer-offline
        - npx ampx pipeline-deploy --branch $AWS_BRANCH --app-id $AWS_APP_ID
```

### Reproduction steps

Given the following schema 
```
const schema = a.schema({
  Account: a.model({
    identity: a.id().required(),
    email: a.email()
  }).identifier(['identity']).authorization(allow => [allow.owner()])
})
```

Add a new model for example : 

```
const schema = a.schema({
  Todo: a
      .model({
        content: a.string(),
      }).authorization(allow => [allow.owner()]),
  Account: a.model({
    identity: a.id().required(),
    email: a.email()
  }).identifier(['identity']).authorization(allow => [allow.owner()])
})
```

Expected is that after deployment using CI/CD Gen 2, the types and everything regarding new model should be defined and written in the `amplify_outputs.json`, while this is not happening (the file get's updated only after a day)

**Comment by scorobogaci** [comment]
I gave attempted to generate the output explicitly using :

`- npx ampx generate outputs --branch $AWS_BRANCH --app-id $AWS_APP_ID`

in the `backend` build like this : 

```
version: 1
backend:
  phases:
    build:
      commands:
        - npm ci --cache .npm --prefer-offline
        - npx ampx pipeline-deploy --branch $AWS_BRANCH --app-id $AWS_APP_ID
        - npx ampx generate outputs --branch $AWS_BRANCH --app-id $AWS_APP_ID
```

still the `amplify_output.json` have not been updated

**Comment by Jay2113** [comment]
Hi @scorobogaci 👋 , thanks for reaching out. I wasn't able to reproduce this issue with the provided data schema. Can you share the following? 
- Amplify app id
- `amplify_outputs.json` file (both the older and the latest). Since these files contain sensitive information, I'd recommend sharing this over a DM on our [community discord server](https://discord.gg/amplify). My username is `Jay Raval`.
- Were you trying to download the latest outputs file from the console when a deployment was is in progress?

**Comment by scorobogaci** [comment]
> Hi @scorobogaci 👋 , thanks for reaching out. I wasn't able to reproduce this issue with the provided data schema. Can you share the following?
> 
> * Amplify app id
> * `amplify_outputs.json` file (both the older and the latest). Since these files contain sensitive information, I'd recommend sharing this over a DM on our [community discord server](https://discord.gg/amplify). My username is `Jay Raval`.
> * Were you trying to download the latest outputs file from the console when a deployment was is in progress?

Hello :)
No, I have downloaded the `amplify_outputs.json` after the deployment was finished. I just added a new model for data, then deployed the stack (Ci triggered automatically) and after checking the output, the new model wasn't there in the `amplify_outputs.json` :(

**Comment by Jay2113** [comment]
@scorobogaci can you share the information requested [here](https://github.com/aws-amplify/amplify-hosting/issues/4039)? Additionally, post a deployment does the AppSync API resource get updated with the new model? Lastly, can you share your entire schema as it will be helpful for us to further investigate into this issue.

**Comment by Jay2113** [comment]
I am going to close this thread since we are unable to reproduce this issue. If you continue to experience it, feel free to create a new issue with the requested information. Thanks.

**Comment by jh83** [comment]
I noticed that this ticket has been closed since it wasn't possible to reproduce the issue.

I found this ticket after spending hours debuging my code before i realized that the "download amplify_outputs.json" gave me a old version. I tried multiple things, refresing the web page etc but the file contents remained old until it suddenly was updated.

**Comment by FraPochetti** [comment]
Thanks for reopening folks!
@Jay2113 Heya, I experienced the same issue.
Discussed with @salihgueler too.

Which files do you need me to share for you to take a closer look?

**Comment by vishaljkk** [comment]
@scorobogaci 

seems like you will have to replace your amplify.yml from your version to mine
Basically we genrate the ouput file first and then do the build
not sure whether it will update the latest output file or not
might be there be other possiblity of adding this command in the end again to update the latest file

// your version
version: 1
backend:
  phases:
    build:
      commands:
        - npm ci --cache .npm --prefer-offline
        - npx ampx pipeline-deploy --branch $AWS_BRANCH --app-id $AWS_APP_ID
        - npx ampx generate outputs --branch $AWS_BRANCH --app-id $AWS_APP_ID

// my version
version: 1
backend:
  phases:
    build:
      commands:
        - npm ci --cache .npm --prefer-offline
        - npx ampx generate outputs --branch $AWS_BRANCH --app-id $AWS_APP_ID        
        - npx ampx pipeline-deploy --branch $AWS_BRANCH --app-id $AWS_APP_ID

**Comment by Jay2113** [comment]
@FraPochetti @scorobogaci @jh83 @vishaljkk Thanks for your continued patience. We have released a fix for this issue. Please let us know if the issue persists where the `download amplify_outputs` button does not appear or if the file is outdated. If it continues, can you share the output of your `npx ampx info` command and also try generating the file directly from the CloudFormation stack via `npx ampx generate outputs --branch $AWS_BRANCH --app-id $AWS_APP_ID`.

**Comment by legendfolk** [comment]
I've encountered this issue, too. I deployed on 5/29 and the amplify_outputs.json was not updated. I left it alone until today 6/1 and I downloaded the amplify_outputs.json from the 5/29 deployment and it was updated. It seems like a backend services synch or cache issue where it is not propagating the updated amplify_outputs.json immediately.

**Comment by JulianOcampo** [comment]
Hi, I'm experiencing the exact same issue.

My Amplify project was initially created without a backend. Later, I added functions and a data schema inside the /amplify folder.

I've already tried:

Switching branches.

Making changes to both backend and frontend code.

Triggering new builds in Amplify Hosting.

Disconnecting and reconnecting the branch.

However, every time I run:
```

npx ampx generate outputs \
  --app-id <my-app-id> \
  --branch main \
  --profile <profile>

```
The generated file always contains only:

```
{
  "version": "1.4"
}
```

<img width="1577" height="737" alt="Image" src="https://github.com/user-attachments/assets/163dc693-3d4b-46c0-9e3b-2d169bae8ee9" />

It seems the backend is never linked to the branch for output generation, even though the resources are visible in the Amplify Console.

Any guidance on how to force Amplify Hosting to detect and associate the backend so that ampx generate outputs returns the full config would be greatly appreciated.

**Comment by nasuadachi** [comment]
This phenomenon is also occurring in my environment.
However, it only occurs in specific branches.
main, test1, and test2 immediately reflect the correct ampliy_outputs.json.
The test-test1 branch ends up with the old version.

The following lines have also been added.
```
npx ampx generate outputs --branch $AWS_BRANCH --app-id $AWS_APP_ID
```
